ts/src/mercado.ts

```diff
@@ -464,7 +465,10 @@ export default class mercado extends Exchange {
                 if (price === undefined) {
                     throw new InvalidOrder (this.id + ' createOrder() requires the price argument with market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount');
                 }
-                request['cost'] = this.priceToPrecision (market['symbol'], amount * price);
+                const amountString = this.numberToString (amount);
+                const priceString = this.numberToString (price);
+                const cost = this.parseNumber (Precise.stringMul (amountString, priceString));
```

Comment:
@samgermain not sure we should use `parseNumber` here because it might return a string, maybe `parseToNumeric` is better?

Reply:
> @samgermain not sure we should use `parseNumber` here because it might return a string, maybe `parseToNumeric` is better?

I've put this snippet on a lot of exchanges, and use parseNumber on all those exchanges, so if I'm going to change it here I would need to go change it in a lot of places

Reply:
@samgermain I think we have to go case by case, as you know parseNumber might return a string if the user changes the number type, but some exchange accept both strings and numbers